### ISSUE TITLE
azurerm_site_recovery_replication_policy application_consistent_snapshot_frequency_in_minutes  needs to support a 0 value

### ISSUE THREAD

**Issue by richshadman** [hashicorp/terraform-provider-azurerm#10067]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.13.3
+ provider registry.terraform.io/hashicorp/azuread v0.10.0
+ provider registry.terraform.io/hashicorp/azurerm v2.39.0
+ provider registry.terraform.io/hashicorp/external v1.2.0
+ provider registry.terraform.io/hashicorp/helm v1.3.0
+ provider registry.terraform.io/hashicorp/kubernetes v1.13.2
+ provider registry.terraform.io/hashicorp/local v1.4.0
+ provider registry.terraform.io/hashicorp/null v2.1.2
+ provider registry.terraform.io/hashicorp/random v2.3.0
+ provider registry.terraform.io/hashicorp/template v2.1.2
+ provider ***/gavinbunney/kubectl v1.6.2

### Affected Resource(s)

* `azurerm_site_recovery_replication_policy`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp

resource "azurerm_site_recovery_replication_policy" "policy" {
  name                                                 = "policy"
  resource_group_name                                  = "resource_group"
  recovery_vault_name                                  = "vault"
  recovery_point_retention_in_minutes                  = 1440
  application_consistent_snapshot_frequency_in_minutes = 0
}
```

### Expected Behaviour

The parameter "application_consistent_snapshot_frequency_in_minutes" needs to support a value of 0, meaning "Off". This is required for setting up a policy to be used with a failover cluster that uses SSD (storage spaces direct).

### Actual Behaviour

The resource fails with an error message stating that the value must be between 1 and xxx.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

See the following from MS: https://azure.microsoft.com/en-us/blog/protect-azure-virtual-machines-using-storage-spaces-direct-with-azure-site-recovery/

I was able to use template deployment and pass a 0 to this value (as well as "crashConsistentFrequencyInMinutes" which your resource does not have a parameter for). This created successfully with a value of "Off" for app consistency. See images below.
![image](https://user-images.githubusercontent.com/59030423/103801873-53ca3800-501c-11eb-9a7e-eef79502e8bb.png)

![image](https://user-images.githubusercontent.com/59030423/103801632-00f08080-501c-11eb-8efa-16dd74dc3e34.png)

### References

* #0000

**Comment by lonegunmanb** [comment]
I think this issue has been solved by #14329 .

**Comment by rcskosir** [comment]
Thanks for opening this issue. This was a problem in the 2.x version of the provider which is no longer actively maintained.
If this is still an issue with the 3.x version of the provider please do let us know by opening a new issue, thanks!
